distsql/select_result.go

```diff
@@ -74,14 +74,32 @@ type selectResult struct {
 	copPlanIDs []fmt.Stringer
 	rootPlanID fmt.Stringer
 
+	fetchErr error
+
 	memTracker *memory.Tracker
 }
 
 func (r *selectResult) Fetch(ctx context.Context) {
 }
 
 func (r *selectResult) fetchResp(ctx context.Context) error {
+	startTime := time.Now()
+	defer func() {
+		if c := recover(); c != nil {
```

Comment:
We need not to do `recover` here since it is in main thread.

Reply:
You mean we could just remove the recovery part and let (if any) OOM panic goes up?

Reply:
Yes.